express-crud/server.ts

```diff
@@ -0,0 +1,151 @@
+import express from 'express';
+import pg from 'pg';
+import { ClientError, errorMiddleware } from './lib';
+
+const app = express();
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false,
+  },
+});
+
+app.use(express.json());
+
+app.get('/api/grades', async (req, res, next) => {
+  try {
+    const sql = `
+  select *
+  from "grades"
+  `;
+    const results = await db.query(sql);
+    const check = results.rows;
+    if (!check) throw new ClientError(404, 'Could not load grades');
```

Comment:
This won't ever happen because if there are no rows results.rows will be an empty array, which is truthy. Besides, when returning all items, you usually return an empty array instead of a 404, so you can simply return results.rows and not bother with this check.